hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/RestoreJob.java

```diff
@@ -32,12 +33,13 @@
 public interface RestoreJob extends Configurable {
   /**
    * Run restore operation
+   * @param targetFileSystem  output file system
    * @param dirPaths          path array of WAL log directories
    * @param fromTables        from tables
    * @param toTables          to tables
    * @param fullBackupRestore full backup restore
    * @throws IOException if running the job fails
    */
-  void run(Path[] dirPaths, TableName[] fromTables, TableName[] toTables, boolean fullBackupRestore)
-    throws IOException;
+  void run(FileSystem targetFileSystem, Path[] dirPaths, TableName[] fromTables,
```

Comment:
nit: looks like dirPaths and fromTables are input related.  targetFileSystem and toTables are output related.  can you actually move targetFileSystem to the 3rd param?